news/_posts/2022-07-01-introducing-horizontal-migration.md

```diff
@@ -0,0 +1,74 @@
+
+Now, rest assured, this wouldn't mean abandoning Twig, Grid, Symfony forms, and CQRS – quite the opposite. Once we had all controllers running on Symfony, we could start the second step: removing Smarty in favor of Twig. If we managed to make HelperList and HelperForm work using Twig, we could easily phase out Smarty. This would simply require refactoring the helpers (in which case we wouldn't need to change any of the controllers), or creating new components that follow the same signature as the old helpers (which would require minimal changes to the controllers). Again, this could be applied everywhere, allowing us to phase out Smarty, the legacy Back office theme, and the legacy layout that envelops Symfony Twig pages.
+
+After that, we could go on to phase out HelperLists in favor of Grids, then HelperForms in favor of Symfony forms and CQRS. And then, we would get a complete migration of the Back office.
+
+We call this approach “horizontal migration”, because it focuses on replacing one layer of legacy components with its modern equivalent across all the pages before moving on to the next layer. In opposition, we call our current approach "vertical migration" because it focuses on migrating all the layers of a page at once. The end result is the same: full migration to Symfony. It’s the path to get there that changes.
+
+## What's next?
+
+We think the horizontal approach provides two clear benefits compared to our current approach:
+- A quicker reduction of the system's overall complexity, because legacy components can be sunset as soon as they are no longer needed, instead of having to wait until the end of the migration to do so.
+- A more gentle learning curve for PrestaShop developers not used to Symfony, who will be able to use HelperList and HelperForm in Symfony controllers.
+
+The main downside of the horizontal approach is that lists and forms need to be migrated twice: once with helpers, then again using the newer components. However, we think its benefits are worth the effort.
+
+The horizontal approach has been [recently approved](https://github.com/PrestaShop/ADR/pull/26) by maintainers, as well as the [first example of a listing page migrated using HelperList](https://github.com/PrestaShop/PrestaShop/pull/27246). This means that this should be the main approach to migration from now on.
+
+The vertical approach is not dead though: it might still be used in pages where the horizontal one would not make tactical sense, either because they are too simple to benefit from the advantages of horizontal migration, or because they require extensive refactoring (e.g. the product page).
```

Comment:
```suggestion
We will keep on working with the vertical approach: it might be used in pages where the horizontal approach would not make tactical sense. Either because they are too simple to benefit from the advantages of horizontal migration, or because they require extensive refactoring (for example, the product page).
```
```suggestion
The vertical approach is not dead though: it might still be used in pages where the horizontal one would not make tactical sense, either because they are too simple to benefit from the advantages of horizontal migration, or because they require extensive refactoring (e.g. the product page).
```

Reply:
Hmm, I think that there's a subtle shift in meaning on your version. I feel that it suggests the idea that it's going to be kept pretty much the same as the horizontal approach, whereas I'm trying to say that we don't intend to abandon it altogether, but we will only use it as an alternative when it makes tactical sense to do so.
```suggestion
The vertical approach is not dead though: it might still be used in pages where the horizontal approach would not make tactical sense, either because the page is too simple to benefit from the advantages of horizontal migration, or because it requires extensive refactoring (for example, the product page).
```

Reply:
Yes this version works for me, great if you could find another word than "dead", if this makes sense technically ;)? Thanks

Reply:
Yes this version works for me, great if you could find another word than "dead", if this makes sense technically ;)? Thanks

Reply:
How about this?
```suggestion
The vertical approach is not completely abandoned, though: it might still be used in pages where the horizontal approach would not make tactical sense, either because the page is too simple to benefit from the advantages of horizontal migration, or because it requires extensive refactoring (for example, the product page).
```